Feature cell fracs

zxkjack123 opened the pull request:
Main changes:

- Add a method `tag_cell_fracs()` to `pyne/pyne/mesh.py:Mesh`. 

- Add a function `test_tag_cell_fracs()` in `pyne/tests/test_mesh.py` to test the adding method

Some white space in `pyne/tests/test_mesh.py` are deleted.

zxkjack123 committed:
20171020_cell_fracs_add

zxkjack123 committed:
20171020_cell_fracs_edit_2

zxkjack123 committed:
20171020_edit_test_mesh.py_3

zxkjack123 committed:
20171023 mesh.py edit 1

zxkjack123 committed:
20171023 mesh.py test_mesh.py nosetest_passed

zxkjack123 committed:
20171023 change according to Paul comments edit 4

zxkjack123 committed:
20171023 mesh.py change according to Paul's comments edit 4

scopatz commented on pyne/mesh.py:
```diff
@@ -1196,6 +1200,52 @@ def cell_fracs_to_mats(self, cell_fracs, cell_mats):
+        voxel_cell_number = np.empty(shape=(num_vol_elements,max_cell_number))
+        voxel_cell_fracs = np.empty(shape=(num_vol_elements,max_cell_number))
+        for ve in range(num_vol_elements):
+            voxel_cell_number[ve] = [int(-1)] * max_cell_number
```
-1 is an integer, you don't need to call `int(-1)`

scopatz commented on pyne/mesh.py:
```diff
@@ -1196,6 +1200,52 @@ def cell_fracs_to_mats(self, cell_fracs, cell_mats):
+        voxel_cell_fracs = np.empty(shape=(num_vol_elements,max_cell_number))
+        for ve in range(num_vol_elements):
+            voxel_cell_number[ve] = [int(-1)] * max_cell_number
+            voxel_cell_fracs[ve] = [float(0.0)] * max_cell_number
```
Similarly 0.0 is a float

scopatz commented on pyne/r2s.py:
```diff
@@ -94,6 +94,10 @@ def irradiation_setup(flux_mesh, cell_mats, alara_params, tally_num=4,
+        # tag cell fracs
+        use_subvoxel = True
+        if use_subvoxel:
+            m.tag_cell_fracs(cell_fracs)
```
These three lines are logically equivalent to only the last line, `m.tag_cell_fracs(cell_fracs)`

scopatz commented:
A couple of minor suggestions @zxkjack123, but thanks for the PR!

zxkjack123 committed:
Update mesh.py

1. `int(-1)` is changed to `-1`
2. `float(0.0)` is changed to `0.0`

zxkjack123 committed:
Update r2s.py

Delete:
``` 
use_subvoxel = True
    if use_subvoxel:
```
Keeps only the last line:
`m.tag_cell_fracs(cell_fracs)`

zxkjack123 commented:
`mesh.py` and `r2s.py` are updated according to the suggestions of @scopatz .

scopatz commented:
Thanks @zxkjack123 - I'll leave this for @gonuke, but it LGTM

zxkjack123 committed:
20171024 edti 5

zxkjack123 committed:
Merge branch 'feature_cell_fracs' of https://github.com/zxkjack123/pyne into feature_cell_fracs

zxkjack123 committed:
20171024 add cell_largest_frac_tag edit5

zxkjack123 committed:
20171025 solve type problem of cell_largest_frac_number int edit 6

gonuke commented on pyne/mesh.py:
```diff
@@ -775,6 +775,14 @@ def __init__(self, mesh=None, structured=False,
+        # add a property: cell_cell_fracs_tag and set it to None
+        self.cell_number_tag = None
+        self.cell_fracs_tag = None
+        self.mat_name_tag = None
```
Should take out the `mat_name_*` tags here since we aren't using them yet.

zxkjack123 replied:
I will delete them in next pull request.

gonuke commented:
Thanks @zxkjack123 - merging now.
